src/sentry/api/bases/project.py

```diff
@@ -144,7 +144,7 @@ def convert_args(
                     self.convert_args.__qualname__, str(organization_slug)
                 ):
                     redirect = redirect.get(
-                        organization__id=organization_slug,
+                        organization__slug__id_or_slug=organization_slug,
```

Comment:
I wonder how this slipped through the tests

Reply:
Looks like I didn't test all the control flows of the convert_args.  😬

Reply:
The redirect only happens when you rename a project.

Reply:
ah I see that's easy to miss